### DISCUSSION TITLE
Block lists

### DISCUSSION THREAD

**Discussion by TheSilentChirp** [AdguardTeam/AdGuardHome#3683] [Q&A]
Does ADguard dedup the block lists?

**Comment by fernvenue** [comment]
> 
> 
> Does ADguard dedup the block lists?

Never encountered it before, and you didn't provide any useful details. I suggest you go here to ask: https://github.com/AdguardTeam/AdGuardHome/discussions

**Comment by ainar-g** [comment]
Hello.  The *rules* within one filtering engine are deduplicated.  Is that what you meant by “deduping the block lists”?

**Reply by TheSilentChirp** [reply]
Example: pfblockerng downloads blocklists from various repositories and then checks for duplicate host names / IPs so that when you query DNS/IP record it does not check twice￼; thus improving performance.

Thanks.

**Reply by ainar-g** [reply]
Yes, the rules from the lists that you add on the “DNS blocklists” page are compiled into a single filtering engine that deduplicates the rules.
